### PR TITLE
[ENH] Rework series plotting to work with numpy arrays

### LINKED ISSUE #2278: [ENH] refactor visualisation/series/_series.py functions
### Describe the feature or idea you want to propose

plot functions for series are all based around pd.Series and pd.DataFrame and do not work with numpy arrays. Should refactor to at least work with numpy and also to default to wide format for multivariate

### Describe your proposed solution

plot_series
plot_lags
plot_correlations
plot_spectrogram

protected functions
_plot_interval
_check_colors

these all require pd.series, should all just work by allowing numpy and internally converting

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TonyBagnall**
Fixes #2278 

the two univariate series plot functions  plot_lag and plot_correlation now work with np.ndarray as well as pd.Series. 

plot_series is the only one that takes multivariate. It was configured to plot either a single series or a list of series, thus making it in effect a duplicate of plot_collection. I have changed this so that it can now only plot a (possibly multivariate) series of valid type, e.g. np.ndarray, pd.Series or pd.DataFrame of shape `(n_channels, n_timepoints)`. It will not work with a collection. 

I have removed testing that was really in effect testing the legacy function check_y.

### CODE REVIEW

File: aeon/visualisation/series/_series.py [hunk 1]
```diff
@@ -66,13 +72,27 @@ def plot_series(
     from matplotlib.cbook import flatten
     from matplotlib.ticker import FuncFormatter, MaxNLocator
 
+    # Convert single multivariate series in wide format to list of series
+    if isinstance(series, pd.DataFrame):
+        series = [row for _, row in series.iterrows()]
+    elif isinstance(series, pd.Series):
+        series = [series]
+    elif isinstance(series, np.ndarray):
```
Comment (baraline) [inline aeon/visualisation/series/_series.py:80]: In this check, don't you need to check if series is not 3d array ?

File: aeon/visualisation/series/tests/test_series_plotting.py [hunk 2]
```diff
@@ -59,64 +44,51 @@ def test_plot_series():
...
-            _plot_series(series_to_plot)
-    else:
-        with pytest.raises(ValueError, match="input must be univariate"):
-            _plot_series(series_to_plot)
+    series_to_plot = "This is a string"
+    with pytest.raises((TypeError), match="found type: <class 'str'>"):
+        plot_series(series_to_plot)
+    series_to_plot = (pd.DataFrame({"y1": y_airline, "y2": y_airline}),)
+    with pytest.raises(ValueError, match="input must be univariate"):
+        plot_series(series_to_plot)
```
Comment (baraline) [inline aeon/visualisation/series/tests/test_series_plotting.py:92]: If 3d is not allowed could bé good to add a check here

File: aeon/transformations/series/_dobin.py [hunk 3]
```diff
@@ -64,18 +64,18 @@ class Dobin(BaseSeriesTransformer):
     >>> import numpy as np
     >>> import pandas as pd
     >>> from aeon.datasets import load_uschange
-    >>> _, X = load_uschange()
-    >>> scaler = MinMaxScaler()
-    >>> X = scaler.fit_transform(X)
+    >>> X = load_uschange()
+    >>> min = MinMaxScaler()
```
Comment (baraline) [inline aeon/transformations/series/_dobin.py:68]: Minor but I would call it minmax rather than just min for clarity

**Review by baraline — COMMENTED** [review]
Minor comment and a check question, otherwise lgtm

**Comment by MatthewMiddlehurst** [comment]
conflicted

File: aeon/visualisation/series/_series.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/visualisation/series/_series.py:80]: its a plot function for a series, but yes a check function its a valid type is a good idea

**Comment by TonyBagnall** [comment]
so its a little more complex than I thought, this function was being used to print lists of pd.Series with different axis, thus allowing two series to have different offsets. AS I have redone it, it is for a single series and I think lists of series should be a separate function.

**Comment by hadifawaz1999** [comment]
mentioning [this](https://github.com/aeon-toolkit/aeon/issues/1036) here as it is related

File: aeon/visualisation/series/_series.py [hunk 4]
```diff
@@ -60,19 +62,39 @@ def plot_series(
...
+    s = []
     for y in series:
-        check_y(y, allow_index_names=True)
-
-    series = list(series)
+        if isinstance(y, np.ndarray):
+            y = y.squeeze()
+            y = pd.Series(y)
+        s.append(y)
+    series = s
```
Comment (baraline) [inline aeon/visualisation/series/_series.py:95]: This felt weird has squeezing shouldn't have any effect here and this code could be replaced by calling pd.series in the if condition for ndarrays above. Otherwise for series/dataframes it does nothing I think ?

**Review by baraline — APPROVED** [review]
Think it is good to only accept single series in this function, so I agree with the changes. Rationalizing the index type checks also looks better. 
Just one non-blocking comment that doesn't impact correctness, feel free to address it or not.

File: aeon/visualisation/series/_series.py (hunk 4 above)
Comment (TonyBagnall) [reply aeon/visualisation/series/_series.py:95]: goof spot is left over from when a list/collection could be passed, so it was previously ok to pass a list of numpy.

**Comment by TonyBagnall** [comment]
> mentioning [this](https://github.com/aeon-toolkit/aeon/issues/1036) here as it is related

good spot, I think this fixes #1036, list of numpy would be with plot_collection

**Review by chrisholder — APPROVED** [review]
